shell/platform/android/io/flutter/embedding/android/KeyboardManager.java

```diff
@@ -0,0 +1,204 @@
+// Copyright 2013 The Flutter Authors. All rights reserved.
+// Use of this source code is governed by a BSD-style license that can be
+// found in the LICENSE file.
+
+package io.flutter.embedding.android;
+
+import android.view.KeyEvent;
+import android.view.View;
+import androidx.annotation.NonNull;
+import io.flutter.Log;
+import io.flutter.embedding.android.KeyboardManager.PrimaryResponder.OnKeyEventHandledCallback;
+import io.flutter.embedding.engine.systemchannels.KeyEventChannel;
+import io.flutter.plugin.editing.TextInputPlugin;
+import java.util.HashSet;
+
+/**
+ * A class to process {@link KeyEvent}s dispatched to a {@link FlutterView}, either from a hardware
+ * keyboard or an IME event.
+ *
+ * <p>A class that sends Android {@link KeyEvent} to the a list of {@link PrimaryResponder}s, and
+ * re-dispatches those not handled by the primary responders.
+ *
+ * <p>Flutter uses asynchronous event handling to avoid blocking the UI thread, but Android requires
+ * that events are handled synchronously. So, when a key event is received by Flutter, it tells
+ * Android synchronously that the key has been handled so that it won't propagate to other
+ * components. Flutter then uses "delayed event synthesis", where it sends the event to the
+ * framework, and if the framework responds that it has not handled the event, then this class
+ * synthesizes a new event to send to Android, without handling it this time.
+ *
+ * <p>A new {@link KeyEvent} sent to a {@link KeyboardManager} can be propagated to 3 different
+ * types of "responder"s (in the listed order):
```

Comment:
Hmm I still prefer "responders" here. I think it's a more established term. "destinations" or "handlers" doesn't sound as good to me since "responders" also conveys that it responds (whether the event is handled), in addition to handling the key event.

Reply:
It's pretty confusing since it's the name for a class. It might not be that expressive either, since technically redispatching to the system does not really respond anything. What about "A new even sent to a KeyboardManager have 3 candidate processors:"?